nodejs/week2/app.js

```diff
@@ -0,0 +1,49 @@
+const express = require("express");
+const data = require("./document.json");
+console.log(data);
+const port = process.env.PORT || 3000;
+const app = express();
+app.use(express.json())
+
+
+app.get('/', (req, res) => {
+ res.send('This is a search engine')
+})
+
+app.get("/search", (req, res) => {
+    const text = req.query.q
+    const document = data.filter(element => Object.values(element).some(value => String(value).includes(text)));
+  if(document.length === 0){ 
+   return res.json({data: "Nothing searched up", message:'Search for something'})}
```

Comment:
I don't think you should handle this case specially here. If there are no results you can just return an empty array.